internal/charmstore/stats.go

```diff
@@ -574,8 +575,8 @@ func (s *Store) IncrementDownloadCountsAsync(id *charm.Reference) {
 
 // IncrementDownloadCounts updates the download statistics for entity id in both
 // the statistics database and the search database.
-func (s *Store) IncrementDownloadCounts(id *charm.Reference) error {
-	key := EntityStatsKey(id, params.StatsArchiveDownload)
+func (s *Store) IncrementDownloadCounts(id *router.ResolvedURL) error {
+	key := EntityStatsKey(&id.URL, params.StatsArchiveDownload)
```

Comment:
We will probably move towards incrementing two stats keys, one for promulgated, the other not.